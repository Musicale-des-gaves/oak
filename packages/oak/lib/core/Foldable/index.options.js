import { DRAG_OPTION } from '../../options';

export default [DRAG_OPTION];
