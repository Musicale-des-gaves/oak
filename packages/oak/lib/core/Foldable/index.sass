@use "colors" as colors

.oak-foldable
  --background-color: var(--oak-background-color)
  --inner-background-color: var(--oak-inner-background-color)
  --drop-zone-color: var(--oak-interaction-color)

  display: flex
  flex-wrap: wrap
  flex-direction: row
  position: relative
  margin-top: 30px
  margin-bottom: 20px
  width: 100%

  .oak-inner
    border-radius: 0 5px 5px 5px

  .oak-drop-zone
    position: absolute
    left: 0
    width: 100%
    z-index: 1

    &:after
      content: ''
      width: 100%
      left: 0
      height: 5px
      border-radius: 2px
      background: var(--drop-zone-color)
      opacity: 0
      position: absolute

    &.oak-before
      top: -35px
      height: 35px

    &.oak-after
      top: 100%
      height: 20px

      &:after
        top: 8px

    &.drag-enter
      &:after
        opacity: 1

  .oak-foldable-content
    padding: 10px
    display: flex
    flex-direction: column
    gap: 20px

  .oak-foldable-section-title
    color: var(--oak-alternate-text-color)

  .oak-foldable-section-content
    padding: 20px
    background: var(--inner-background-color)
    display: flex
    flex-direction: column
    align-items: center
    min-height: 100px
    height: fit-content
    border-radius: 5px

    & :nth-child(2)
      &.oak-element
        margin-top: 20px

    &.drag-enter .oak-foldable-content-empty
      background: var(--drop-zone-color)
      border-radius: 5px

      .oak-handle
        display: none

    .oak-handle 
      width: 26px
      height: 26px

      .oak-handle-inner i
        font-size: 20px 

    .oak-foldable-section-content-empty
      width: 100%
      display: flex
      justify-content: center
      align-items: center
      min-height: 80px

      .oak-handle 
        width: 40px
        height: 40px

        .oak-handle-inner i
          font-size: 30px

    .oak-element
      width: 100%

      &.oak-row
        margin-top: 20px

  > .oak-options
    top: -25px
    left: 0
    transform: none
    display: flex
    visibility: visible
    z-index: 2

    a.oak-option
      &:first-child
        border-radius: 3px 0 0 0

        &:last-child
          border-radius: 3px 3px 0 0

      &:last-child, &.oak-edit
        border-radius: 0 3px 0 0

      &.oak-drag-handle
        cursor: move
        cursor: grab

        &:active
          cursor: gragging
