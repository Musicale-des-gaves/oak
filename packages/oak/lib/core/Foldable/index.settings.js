export default {
  title: t => t('core.components.foldable.settings.title', 'Foldable options'),
  floatingSettings: {
    placement: 'right-start',
    autoPlacement: {
      alignment: 'start',
    },
  },
  fields: [{
    type: 'select',
    key: 'settings.seeMorePosition',
    default: 'after',
    label: t => t('core.components.foldable.settings.seeMorePosition.title',
      'See more placement'
    ),
    options: [{
      title: t => t('core.components.foldable.settings.seeMorePosition.before',
        'Before'
      ),
      value: 'before',
    }, {
      title: t => t('core.components.foldable.settings.seeMorePosition.after',
        'After'
      ),
      value: 'after',
    }],
  }],
};
