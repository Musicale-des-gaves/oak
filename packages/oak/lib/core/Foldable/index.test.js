import { fireEvent, render, waitFor } from '@testing-library/react';
import { withBuilder } from '@tests-utils';

import Foldable from '.';

describe('<Foldable />', () => {
  it('should render', () => {
    const { container, unmount } = render(withBuilder((
      <Foldable
        element={{
          content: [],
          seeMore: [],
          seeLess: [],
        }}
      />
    )));

    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should render as many content as given on each section', () => {
    const { container, rerender, unmount } = render(withBuilder(
      <Foldable
        element={{
          content: [
            { type: 'row', cols: [], id: 1 },
            { type: 'row', cols: [], id: 2 },
            { type: 'row', cols: [], id: 3 },
          ],
          seeMore: [],
          seeLess: [],
        }}
      />
    ));

    expect(container).toMatchSnapshot('3 rows');

    rerender(withBuilder(
      <Foldable
        element={{
          content: [],
          seeMore: [
            { type: 'row', cols: [], id: 1 },
            { type: 'row', cols: [], id: 2 },
          ],
          seeLess: [],
        }}
      />
    ));

    expect(container).toMatchSnapshot('2 rows');

    rerender(withBuilder(
      <Foldable
        element={{
          content: [],
          seeMore: [],
          seeLess: [
            { type: 'row', cols: [], id: 1 },
            { type: 'row', cols: [], id: 2 },
            { type: 'row', cols: [], id: 3 },
            { type: 'row', cols: [], id: 4 },
          ],
        }}
      />
    ));

    expect(container).toMatchSnapshot('4 rows');

    unmount();
  });

  it('should display one catalog if content is empty and two if not', () => {
    const { container, rerender, unmount } = render(withBuilder(
      <Foldable
        element={{
          content: [],
          seeMore: [],
          seeLess: [],
        }}
      />
    ));

    expect(container).toMatchSnapshot('No content');

    rerender(withBuilder(
      <Foldable
        element={{
          content: [{ type: 'row', cols: [], id: 1 }],
          seeMore: [{ type: 'row', cols: [], id: 2 }],
          seeLess: [{ type: 'row', cols: [], id: 3 }],
        }}
      />
    ));

    expect(container).toMatchSnapshot('With content');

    unmount();
  });

  it('should remove the correct content', async () => {
    const mockRemoveElement = jest.fn();
    const { container, unmount } = render(withBuilder(
      <Foldable
        element={{
          content: [
            { type: 'row', cols: [], id: 1 },
            { type: 'row', cols: [], id: 2 },
            { type: 'row', cols: [], id: 3 },
            { type: 'row', cols: [], id: 4 },
          ],
          seeMore: [],
          seeLess: [],
        }}
      />
      , { removeElement: mockRemoveElement }));

    const content = container
      .querySelectorAll('.oak-foldable-section-content')[2];
    const contentToDelete = { type: 'row', cols: [], id: 3 };
    expect(content.querySelectorAll('.oak-element').length).toEqual(4);

    fireEvent.click(content.querySelectorAll('.oak-remove')[2]);
    await waitFor(() => {
      expect(mockRemoveElement)
        .toHaveBeenCalledWith(contentToDelete, expect.any(Object))
    });

    unmount();
  });

  it('should add the right element', async () => {
    const mockConstruct = jest.fn().mockReturnValue({
      type: 'my-component',
      settings: {},
      content: [],
    });
    const componentMock = {
      id: 'first-component',
      name: 'component 1',
      construct: mockConstruct,
    };
    const groupMock = [{
      type: 'group',
      id: 'group1',
      name: 'group 1',
      components: [],
    }, {
      group: 'group1',
      component: componentMock,
    }];
    const element = {
      content: [{ type: 'text', content: [], id: 1 }],
      seeMore: [],
      seeLess: [],
    };
    const mockAddElement = jest.fn();
    const { container, rerender, unmount } = render(withBuilder((
      <Foldable element={element} />
    ), { addElement: mockAddElement, components: groupMock }));

    let content = container
      .querySelectorAll('.oak-foldable-section-content')[2];

    fireEvent.click(content.querySelector('.oak-catalogue .oak-handle'));
    await waitFor(() => {
      expect(
        content.querySelector('.oak-popover .oak-components')
      ).toBeTruthy();
    });

    content = container.querySelectorAll('.oak-foldable-section-content')[2];

    let popover = content.querySelector('.oak-popover .oak-components');
    fireEvent.click(popover.querySelector('.oak-components .oak-component a'));

    await waitFor(() => {
      expect(mockAddElement).toHaveBeenCalledWith({
        type: 'my-component',
        settings: {},
        content: [],
      }, { parent: element.content, position: 'before' });
    });

    rerender(withBuilder((
      <Foldable element={element} />
    ), { addElement: mockAddElement, components: groupMock }));

    content = container.querySelectorAll('.oak-foldable-section-content')[2];

    fireEvent.click(content.querySelectorAll('.oak-catalogue .oak-handle')[1]);
    await waitFor(() => {
      expect(
        content.querySelector('.oak-popover .oak-components')
      ).toBeTruthy();
    });

    popover = content.querySelector('.oak-popover .oak-components');
    fireEvent.click(popover.querySelector('.oak-components .oak-component a'));

    await waitFor(() => {
      expect(mockAddElement).toHaveBeenCalledWith({
        type: 'my-component',
        settings: {},
        content: [],
      }, { parent: element.content, position: 'after' });
    });

    expect(mockConstruct).toHaveBeenCalledTimes(2);
    unmount();
  });

  it('should add content on the right section', async () => {
    const mockConstruct = jest.fn().mockReturnValue({
      type: 'my-component',
      settings: {},
      content: [],
    });
    const componentMock = {
      id: 'first-component',
      name: 'component 1',
      construct: mockConstruct,
    };
    const groupMock = [{
      type: 'group',
      id: 'group1',
      name: 'group 1',
      components: [],
    }, {
      group: 'group1',
      component: componentMock,
    }];
    const element = {
      content: [{ type: 'text', content: [], id: 1 }],
      seeMore: [{ type: 'button', content: [], id: 1 }],
      seeLess: [{ type: 'title', content: [], id: 1 }],
    };
    const mockAddElement = jest.fn();
    const { container, unmount } = render(withBuilder((
      <Foldable element={element} />
    ), { addElement: mockAddElement, components: groupMock }));

    //adding on content
    let content = container
      .querySelectorAll('.oak-foldable-section-content')[2];

    fireEvent.click(content.querySelector('.oak-catalogue .oak-handle'));
    await waitFor(() => {
      expect(
        content.querySelector('.oak-popover .oak-components')
      ).toBeTruthy();
    });

    content = container.querySelectorAll('.oak-foldable-section-content')[2];

    let popover = content.querySelector('.oak-popover .oak-components');
    fireEvent.click(popover.querySelector('.oak-components .oak-component a'));

    await waitFor(() => {
      expect(mockAddElement).toHaveBeenCalledWith({
        type: 'my-component',
        settings: {},
        content: [],
      }, { parent: element.content, position: 'before' });
    });

    // adding on see more
    content = container.querySelectorAll('.oak-foldable-section-content')[0];

    fireEvent.click(content.querySelector('.oak-catalogue .oak-handle'));
    await waitFor(() => {
      expect(
        content.querySelector('.oak-popover .oak-components')
      ).toBeTruthy();
    });

    content = container.querySelectorAll('.oak-foldable-section-content')[0];

    popover = content.querySelector('.oak-popover .oak-components');
    fireEvent.click(popover.querySelector('.oak-components .oak-component a'));

    await waitFor(() => {
      expect(mockAddElement).toHaveBeenCalledWith({
        type: 'my-component',
        settings: {},
        content: [],
      }, { parent: element.seeMore, position: 'before' });
    });

    // adding on see less
    content = container.querySelectorAll('.oak-foldable-section-content')[1];

    fireEvent.click(content.querySelector('.oak-catalogue .oak-handle'));
    await waitFor(() => {
      expect(
        content.querySelector('.oak-popover .oak-components')
      ).toBeTruthy();
    });

    content = container.querySelectorAll('.oak-foldable-section-content')[1];

    popover = content.querySelector('.oak-popover .oak-components');
    fireEvent.click(popover.querySelector('.oak-components .oak-component a'));

    await waitFor(() => {
      expect(mockAddElement).toHaveBeenCalledWith({
        type: 'my-component',
        settings: {},
        content: [],
      }, { parent: element.seeLess, position: 'before' });
    });
    unmount();
  });
});
