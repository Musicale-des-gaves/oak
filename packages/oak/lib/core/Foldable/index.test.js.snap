// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Foldable /> should display one catalog if content is empty and two if not: No content 1`] = `
<div>
  <div>
    <div
      class="oak-drop-zone oak-before"
    />
    <div
      class="oak-foldable-content"
    >
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when collapsed
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when expanded
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Content
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="oak-drop-zone oak-after"
    />
  </div>
</div>
`;

exports[`<Foldable /> should display one catalog if content is empty and two if not: With content 1`] = `
<div>
  <div>
    <div
      class="oak-drop-zone oak-before"
    />
    <div
      class="oak-foldable-content"
    >
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when collapsed
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="2"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":2}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when expanded
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="3"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":3}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Content
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="1"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":1}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="oak-drop-zone oak-after"
    />
  </div>
</div>
`;

exports[`<Foldable /> should render 1`] = `
<div>
  <div>
    <div
      class="oak-drop-zone oak-before"
    />
    <div
      class="oak-foldable-content"
    >
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when collapsed
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when expanded
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Content
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="oak-drop-zone oak-after"
    />
  </div>
</div>
`;

exports[`<Foldable /> should render as many content as given on each section: 2 rows 1`] = `
<div>
  <div>
    <div
      class="oak-drop-zone oak-before"
    />
    <div
      class="oak-foldable-content"
    >
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when collapsed
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="1"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":1}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="2"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":2}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when expanded
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Content
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="oak-drop-zone oak-after"
    />
  </div>
</div>
`;

exports[`<Foldable /> should render as many content as given on each section: 3 rows 1`] = `
<div>
  <div>
    <div
      class="oak-drop-zone oak-before"
    />
    <div
      class="oak-foldable-content"
    >
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when collapsed
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when expanded
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Content
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="1"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":1}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="2"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":2}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="3"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":3}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="oak-drop-zone oak-after"
    />
  </div>
</div>
`;

exports[`<Foldable /> should render as many content as given on each section: 4 rows 1`] = `
<div>
  <div>
    <div
      class="oak-drop-zone oak-before"
    />
    <div
      class="oak-foldable-content"
    >
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when collapsed
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Label when expanded
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="1"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":1}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="2"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":2}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="3"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":3}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-element oak-row"
            draggable="false"
            id="4"
          >
            <div
              class="oak-inner"
            >
              <pre
                class="oak-unknown oak-element-content-inner"
              >
                {"type":"row","cols":[],"id":4}
              </pre>
            </div>
            <div
              class="oak-options"
            >
              <a
                class="oak-option oak-remove"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  clear
                </i>
              </a>
              <a
                class="oak-option oak-duplicate"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_copy
                </i>
              </a>
              <a
                class="oak-option oak-copy"
                draggable="false"
                href="#"
              >
                <i
                  class="oak-icons"
                >
                  content_paste
                </i>
              </a>
            </div>
          </div>
          <div
            class="oak-catalogue"
          >
            <a
              class="oak-handle"
              draggable="false"
            >
              <span
                class="oak-handle-inner"
              >
                <i
                  class="oak-icons"
                >
                  add
                </i>
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="oak-foldable-section"
      >
        <div
          class="oak-foldable-section-title"
        >
          Content
        </div>
        <div
          class="oak-foldable-section-content"
        >
          <div
            class="oak-foldable-section-content-empty"
          >
            <div
              class="oak-catalogue"
            >
              <a
                class="oak-handle"
                draggable="false"
              >
                <span
                  class="oak-handle-inner"
                >
                  <i
                    class="oak-icons"
                  >
                    add
                  </i>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="oak-drop-zone oak-after"
    />
  </div>
</div>
`;
